sbi/neural_nets/estimators/score_estimator.py

```diff
@@ -537,7 +537,7 @@ def drift_fn(self, input: Tensor, times: Tensor) -> Tensor:
         Returns:
             Drift function at a given time.
         """
-        phi = -0.5 * self._beta_schedule(times)
+        phi = -0.5 * self._beta_schedule(times).to(input.device)
```

Comment:
Not sure if why this is required i.e. shouldn't be "time" on the right device? If this would cause problems, then standard sampling also should not have worked.

Correct device is ensured [here](https://github.com/sbi-dev/sbi/blob/333427f2dd71cbcc2b3d08c6702c088463b19901/sbi/samplers/score/predictors.py#L115), which explains why standard sampling did work.

But it anyway is better to ensure that the correct all methods output the correct device as done now explicitly. So I think these lines could also be removed now, right?

But I see

Reply:
indeed, I fixed the device handling step by step and this was required at some point. but I was able to remove it now probably because I fixed it at a higher level afterwards.